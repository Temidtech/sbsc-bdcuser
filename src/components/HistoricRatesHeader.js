import React, { PureComponent } from 'react';
import { View, Text, Image } from 'react-native';
import { HistoricRatesStyles as styles } from '../styles';
import Images from '@assets/images';

export default class HistoricRatesHeader extends PureComponent {
  render() {
    return (
      <View style={styles.tableHeading}>
        <View style={styles.dateColumn}>
          <Text style={styles.headerTextRow}>Date</Text>
        </View>
        <View style={styles.currencyColumn}>
          <Text style={styles.headerTextRow}>Currency</Text>
        </View>
        <View style={styles.buyRangeColumn}>
          <Image source={Images.ngr} style={styles.headerImage} />
          <Text style={styles.headerTextRow}>Buy Range</Text>
        </View>
        <View style={styles.sellRangeColumn}>
          <Image source={Images.ngr} style={styles.headerImage} />
          <Text style={styles.headerTextRow}>Sell Range</Text>
        </View>
      </View>
    );
  }
}
