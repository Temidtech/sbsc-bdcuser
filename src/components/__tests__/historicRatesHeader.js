import React from 'react';
import renderer from 'react-test-renderer';
import HistoricRatesHeader from '../HistoricRatesHeader';

describe('<HistoricRatesHeader />', () => {

  it('renders the HistoricRatesHeader component correctly', () => {
    const tree = renderer.create(
      <HistoricRatesHeader />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
