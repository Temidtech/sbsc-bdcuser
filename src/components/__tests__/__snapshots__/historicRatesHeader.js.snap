// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoricRatesHeader /> renders the HistoricRatesHeader component correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "width": "30%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Italic",
          "fontSize": 13,
        }
      }
    >
      Date
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "20%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Italic",
          "fontSize": 13,
        }
      }
    >
      Currency
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 0,
        "paddingLeft": 0,
        "width": "25%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/img/ngr.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 5,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Italic",
          "fontSize": 13,
        }
      }
    >
      Buy Rate
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "25%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/img/ngr.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 5,
          "width": 20,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontFamily": "OpenSans-Italic",
          "fontSize": 13,
        }
      }
    >
      Sell Rate
    </Text>
  </View>
</View>
`;
